### PR TITLE
Improved internal model support

### PR DESCRIPTION
**Pull request by agonza40**
### This project uses [semver](semver.org), please check the scope of this pr:
 - [ ] #none# - documentation fixes and/or test additions
 - [x] #patch# - backwards-compatible bug fix
 - [ ] #minor# - adding functionality in a backwards-compatible manner
 - [ ] #major# - incompatible API change

# CHANGELOG
**Refactored** handling of view specified models
**Added** support for view specified models in arrays.

### CODE REVIEW

File: src/normalize-model-and-view.js [hunk 1]
```diff
@@ -25,6 +27,27 @@ export function getModelPath (reference) {
   return path
 }
 
+function appendModelPath (modelPath, id, internal) {
```
Comment (sophypal) [inline src/normalize-model-and-view.js]: You know what I'm gonna say right. jsdoc :)

File: src/normalize-model-and-view.js [hunk 2]
```diff
@@ -25,6 +27,27 @@ export function getModelPath (reference) {
...
+  if (modelPath === '') {
+    return addedModelPath
+  }
+  return `${modelPath}.${addedModelPath}`
+}
+
+function expandCell (cell, cellDefinitions) {
+  cell = _.clone(cell)
+  while (cell.extends) {
+    const extendedCell = cellDefinitions[cell.extends]
```
Comment (sophypal) [inline src/normalize-model-and-view.js]: Is there prior validation that this `extends` is valid? I come across invalid schema all the time due to mispelling and hate it when I have to dig through the source to figure out it was just a mispelling.

File: src/normalize-model-and-view.js (hunk 1 above)
Comment (agonza40) [reply src/normalize-model-and-view.js]: Gah, how could I forget 😅

**Comment by sophypal** [comment]
👍 

[Approved with PullApprove](https://pullapprove.com/ciena-blueplanet/bunsen-core/pull-request/107/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=ciena-blueplanet/bunsen-core)

File: src/normalize-model-and-view.js (hunk 2 above)
Comment (agonza40) [reply src/normalize-model-and-view.js]: That's a good idea. I'll open an issue for it, and for now I'll put some error throwing code that will at least making it obvious when that is the problem for now.

**Comment by sophypal** [comment]
👍 

[Approved with PullApprove](https://pullapprove.com/ciena-blueplanet/bunsen-core/pull-request/107/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=ciena-blueplanet/bunsen-core)

**Pull request merged**